### PR TITLE
transformations: Remove target argument from stencil conversion

### PR DESCRIPTION
**Pull request by PapyChacal**
No need to specialize.
This was done solely to map the fastest varying dimension to the right CUDA dimension; now done by xDSL's inhouse `gpu-map-parallel-loops` in a simpler and independent manner :slightly_smiling_face:

### CODE REVIEW

File: tests/filecheck/dialects/stencil/hdiff_gpu.mlir
Comment (georgebisbas) [inline tests/filecheck/dialects/stencil/hdiff_gpu.mlir:1]: Why is this totally dropped?

Comment (PapyChacal) [reply tests/filecheck/dialects/stencil/hdiff_gpu.mlir:1]: It was testing the gpu lowering; this removes the GPU lowering

**Comment by PapyChacal** [comment]
> Looks good, do we expect any perf diff here?

No, that's why I implemt `gpu-map-parallel-loops` in xDSL.
This is sufficient to end up generating the same CUDA code.

**Review by AntonLydike — APPROVED** [review]
Noice!

**Pull request merged**